test/rekt/features/sequence/feature.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sequence
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/test"
+	"github.com/google/uuid"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/eventing/test/rekt/resources/pingsource"
+	"knative.dev/eventing/test/rekt/resources/sequence"
+
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/manifest"
+	"knative.dev/reconciler-test/pkg/resources/service"
+)
+
+func SequenceTest() *feature.Feature {
+	f := feature.NewFeatureNamed("Sequence test.")
+
+	sequenceName := feature.MakeRandomK8sName("sequence")
+	source := feature.MakeRandomK8sName("source")
+	sink := feature.MakeRandomK8sName("sink")
+	// Sequence's steps
+	step1 := feature.MakeRandomK8sName("step1")
+	step2 := feature.MakeRandomK8sName("step2")
+	step3 := feature.MakeRandomK8sName("step3")
+	msgAppender1 := "-step1"
+	msgAppender2 := "-step2"
+	msgAppender3 := "-step3"
+
+	channelTemplate := sequence.ChannelTemplate{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: "messaging.knative.dev/v1",
+			Kind:       "InMemoryChannel",
+		},
+	}
```

Comment:
Can we take the template as parameter so that we can use the same feature for other channels?

Reply:
Sure

Reply:
I 've encapsulated the IMChannel for subsequent calls